### PR TITLE
Add Trino 382 release notes

### LINKED ISSUE #12425: Release notes for 382

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for the upcoming Trino 382 release.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Documentation only.

> How would you describe this change to a non-technical end user or system administrator?

Add the release notes for Trino 382 with sections for general, security, connectors and other topics as included in the release. 

## Related issues, pull requests, and links

* Fixes #12425

## Documentation

( ) No documentation is needed.
(✅ ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅ ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

##  Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌  rn ✅ / ❌  docs
✅  rn - release note added and verified, or assessed to be not necessary, set to ❌ RN before completion
✅  docs - need for docs assessed and merge, or assessed to be not necessary, set to ❌ docs before completion

## 16 May 2022 ✅  

* https://github.com/trinodb/trino/commit/28ffbd7262fc9a2e93312d5a04497ea4dc2bcb97 @dain #12395 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/d7777f9cade37fc75dfcf1aba90a13025db596bb @ebyhr #12302 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7051ef19b615d141176411e2cf6edebc7adde33b @ebyhr #12302 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/acfce2a113dd1f63b88accd01353aca991170dee @ebyhr #12302 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/2d532b28499477146b7a79e9792898b817509c50 @electrum #12417 ✅ rn ✅ docs

## 17 May 2022 ✅ 

* https://github.com/trinodb/trino/commit/52464200cb21320fcecfa34416afc981b8f4f55a @findepi  #12411 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/9583a8d24cb53168d1224dadcb395e4331e22739 @findepi #12384 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/264a66f92a139a7e0be117d81f3563b7def6f16d @ebyhr #12377 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/390d4aaacb8af90e9ff957cbd9f9a70c11046eba @ebyhr #12377 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a71e12c690c5cdb36a98318449c61f8942d0bac2 @findepi #12334 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a71e12c690c5cdb36a98318449c61f8942d0bac2 @findepi #12334 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/8f0dc103a6238be01b38ac8329971be220322973 @findepi #12420 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/37adec90be8364db40dcce9f54995e688741e110 @ebyhr #12374  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/50582d295a91b44f955e69fed83eae0272c87a0a @ebyhr #12401  ✅   rn ✅ docs
* https://github.com/trinodb/trino/commit/42ed1bd48578ef5f8d3e4e4f6a38278f5e6d867e @ebyhr #12451 ✅ rn ✅ docs

## 18 May 2022 ✅   

* https://github.com/trinodb/trino/commit/ea31fdcd30670b705c9ea02b7fff57a7d1543193 @losipiuk #12445  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/ec406c154388ee10d8471271e8f8bba6da3cb61f @findepi #12379 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/cc9f226b132bd72bb56d8ade3c0eaa7d4ab4a7c8 @findepi #12379 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/2ff665b83789d0714c7e7cd5489956468e6d2677 @findepi #12379 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/be260e5a383a0894722cd31832cb35e94f297475 @findepi #12379 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/c92ea3b3b61acfd4ab972cd331cdf871ff185ba5 @sopel39 #12257  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/320250ecfe38f242a1c9866b274f6f4b3d999959 @losipiuk #12435 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/dda4d2c37a19425d21bd2b4004e2b9af0b1e4a36 @losipiuk #12435 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/119c6bd32bf85572563ea2cf1e9e5d2ebb32fa28 @losipiuk #12435 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/1d51705fdd87ec2a2d72694341da2ad8eef5001f @ebyhr #12408 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a958faae09d80f9baac5c96ac20bf2bb6dfdbcf4 @ebyhr #12408 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/eae834407bcbbb6413d869cbdb606b0559d20fbc @dain #12396 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/013eb28e65ce7974d296779ad9564ae9b6ca511c @ebyhr #12471 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/543ae143cadeb47ab03af4197dae9d00ff5baf7c @hashhar #12446  ✅ rn ✅ docs

## 19 May 2022 ✅ 

* https://github.com/trinodb/trino/commit/78661f707a3ecb85f41b57f699c4da1d95f1f5f2 @findepi #12460 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/223a11b90b2fa58d111c83070196182aefc062a6 @losipiuk #12478 ✅   rn ✅ docs
* https://github.com/trinodb/trino/commit/21c6af128f16a974a0b4ecad0823d142403f6054 @ebyhr #12344  ✅ rn ✅   docs, in discussion with @claudiusli and @colebow 
* https://github.com/trinodb/trino/commit/ac079a43d1cc0e198ffb63d89a284244a3fe719b @ebyhr ✅ rn ✅ docs

## 20 May 2022 ✅ 

* https://github.com/trinodb/trino/commit/95499573d42a07120ca95715498c0857417b6f98 @ebyhr #12448 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/d2cbe9d0e0c13a2bdfb1dcf1adfd9aa222955132 @findepi #12459 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/f8ae2ac859125520a8b403968b39533274b79461 @findepi #12459 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/9be3b8d1efb293c56c7fe7d3a065f35f16ae3845 @findepi #12468 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/acf3c9c867c85edc2f287ac56181909506d32284 @findepi  #12468 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/29770ae372758c78fdbba6fa9cfe3781eaa2542a @electrum #12356 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/b533974b5e6fc4c19d38c75021720287fa0458c2 @electrum #12438  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/f2ab55b985a7251a97b1db501a07b7a18a2cdd25 @electrum #12387 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/11731fc4921b7fae90ca1ce2ae2953ecd83601f0 @electrum #11564 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/9afbb73bcfa189a997b4d4be239c62fdbf3743a4 @kasiafi  #12488 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/2c45353d843ce1840ec5378387c44b599b372504 @findepi #12487 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7b487a3bdd3c27e38f57b0bb13e1001a97b46f74 @electrum  #12421 ✅ rn ✅ docs

## 21 May 2022 ✅   

* https://github.com/trinodb/trino/commit/3b9e47d7bac845e289f808c5f04c1e6d54f64c35 @losipiuk #12311 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/cb1889063b1d42d42728d372c01c0e1529789eb4 @losipiuk #12311 ✅ rn ✅ docs

## 22 May 2022 ✅ 

* https://github.com/trinodb/trino/commit/be2b53c83bcf40faf523ac10c49f28b63509d0fb @ebyhr #12505 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a398d66aa57ba14b61c5a208d9afc4f62c832ec6 @ebyhr  #12428 ✅ rn ✅ docs

## 23 May 2022 ✅  

* https://github.com/trinodb/trino/commit/e235c9953fefa0602057ba82355e15c18430505f @ebyhr  #12458  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/5d0b73a34f14b18e03f57d56b27cbf844a6c01de @ebyhr #12458  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/c750f4e2fa83a939f8b394a0f1f1457d0f453f89 @findepi #12439 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/dacd3023710ee00cdbc86cf483d531cb482f64c0 @findepi #12439  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/86db89c8affce414fc4aeb52020766380aeec9fe @findepi #12259 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/d292dedc02b4e2db7c089e2d7629e6c57067206b @kasiafi #12476  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/239d84791963cb6982c9d82408ed9b42da795f73 @findepi #12515  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7c69c7dd226ac8d34d5ea0c26acc20209f5f3011 @ebyhr #12483  ✅ rn ✅ docs

## 24 May 2022 ❌   

* https://github.com/trinodb/trino/commit/319d40ed0a23fb42e513ab679e1094010d25c9d2 @findepi #12517 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/e5874ef69ab389466f08a3ed74b076753049cc42 @findepi #11565 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/997820118e609bfcd51c0081fb72daec2927e461 @findepi #11565 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/a3936c90caed36bec1ccd46d4ae32962bd098857 @findepi #11565 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/9532493ebf1a34244c9e6e19c4866080606e496b @findepi #11565 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/4338264e170e0a63770460ca6acd23a18cff4988 @findepi #11565 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/465137da7441e88bc65235a0a622b8240669c7be @findepi #12400 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/f365a80bb3d5b37614a6c8cc457a892f824fd94a @findepi #12400 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/0da6e0101d67f1a40372f766030db22ba5ecd422 @ebyhr #12371 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/f7c64e508d7ebe8905ba65f04230af26dbac262c @findepi #11845 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/762d90bfc80426f82e37a172e1408b4588822fe7 @findepi #11845 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/7834be9b981626ec63ad6a4de692c88bd3c69688 @findepi #12343 ✅ rn ❌  docs, https://github.com/trinodb/trino/pull/12534
* https://github.com/trinodb/trino/commit/2d8fbb8028ef1b371a3f253d27d3132e1cf8222a @findepi #11732 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/755dc914198284b1456811a216d7230634444e48 @findepi #11732 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/b89aac68c43e5392f23b8d6ba053bbeb6df85028 @findepi #11732 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/ab25d52b4f54cd9be3973fc8a2fadd99bacc73f4 @arhimondr #12360 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/1137a9e3f2dda98c224de19057d2d19fdf2575b2 @arhimondr #12360 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/2726bdfe3fe29334b4b65fe3b03b9f86c02b896f @arhimondr #12360 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/fcfd210880778775738c4ec5ce447be7a9db9962 @arhimondr #12472 ✅ rn ✅ docs

## 25 May 2022 ❌  

* https://github.com/trinodb/trino/commit/ba032b73ec72dd66a908861d5c2a9bb23c59829d @kasiafi #12531  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/6030c40c929950a4fae8485fb32ab6c5f45a91be @kasiafi #12531  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/be369d0099964066baff6df1f41506d208eeddea @findepi  #12539  ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/575ecc16b7c603ea632747c57c00669aafc9a6c6 @losipiuk #12549 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/366f81efadc27be434d98f35511f39d5424cb0ce @findepi #12416 ❌ rn ✅ docs
* https://github.com/trinodb/trino/commit/db099cf12fc1e4bb5473d36bab822ce8286a2b79 @findepi #12416 ❌ rn ✅ docs
* https://github.com/trinodb/trino/commit/c58d9ea94ec481a0ba9e543f94f9be9dade2f0a0 @findepi #12527 ✅ rn ✅ docs
* https://github.com/trinodb/trino/commit/9ddaa60df18f74247d299cf0ce4c81d9383cfb9d @findepi #12197 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-382.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+## Druid connector
+
+## Elasticsearch connector
+
+## Google Sheets connector
+
+## Hive connector
+
+* Fix failure when ACID table column name collides with ACID internal names. ({issue}`12401`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-382.md:46]: @ebyhr can you explain more .. as it stands I am not sure I can make sense of it

File: docs/src/main/sphinx/release/release-382.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# Release 382 (25 May 2022)
+
+## General
+
+* Improve join performance. ({issue}`12257`)
+* Improve coordinator memory management for task-based fault-tolerant execution.
+  This decreases the chance of queries failing with the
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-382.md:7]: @losipiuk  ... what do you think.. could we just remove the sentence here?

Comment (losipiuk) [reply docs/src/main/sphinx/release/release-382.md:7]: Depending on how explicit we want to be in RN. With the extra sentence, it is easier for users to understand if the new release will solve the problems they are seeing or not. But I do not feel strongly.

File: docs/src/main/sphinx/release/release-382.md (hunk 1 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-382.md:49]: Previously, query failed when ACID table contains ACID column names (e.g. `operation`)

File: docs/src/main/sphinx/release/release-382.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+* Allow metastore caching without caching partition-level information. This can be enabled using
+  `hive.metastore-cache.cache-partitions` catalog configuration property. ({issue}`12343`)
+* Fix queries sometimes failing with `NoSuchElementException: No value present` when metastore caching
+  is enabled. ({issue}`12513`)
+* Fix failure when ACID table column name collides with ACID internal names. ({issue}`12401`)
+
+## Iceberg connector
+
+* Support updating Iceberg table partitioning using `ALTER TABLE ... SET PROPERTIES`. ({issue}`12174`)
+* Add support for Bloom filters when using ORC file format. ({issue}`11732`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-382.md:60]: fyi @findepi .. changed issue number here..

File: docs/src/main/sphinx/release/release-382.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+* Rename the `delete_orphan_files` table procedure to `remove_orphan_files`. ({issue}`12468`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-382.md:61]: @findepi ... minor reword here ..

File: docs/src/main/sphinx/release/release-382.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
+# Release 382 (25 May 2022)
+
+## General
+
+* Add support for fault-tolerant execution with [exchange spooling on Google Cloud Storage](fte-exchange-manager). ({issue}`12360`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-382.md:5]: @arhimondr @linzebing .. fyi .. both reworded a bit ..  I will work with @colebow and @jhlodin to get a follow up doc PR off the ground with some minor tweaks

File: docs/src/main/sphinx/release/release-382.md (hunk 1 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-382.md:30]: > you mean columns potentially used some sort of keywords as columns names and that would result in query failures? 

Yes. The column names are operation, originalTransaction, bucket, rowId, row & currentTransaction.

File: docs/src/main/sphinx/release/release-382.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+* Drop support for exchange spooling on S3 with for the legacy schemes `s3n://` and `s3a://`. ({issue}`12360`)
+* Improve join performance. ({issue}`12257`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-382.md:7]: Maybe add: "... when one of the tables is small" or something to that effect.

File: docs/src/main/sphinx/release/release-382.md (hunk 2 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-382.md:9]: I would rephrase as:

> Fix potential query failures due to `EXCEEDED_TASK_DESCRIPTOR_STORAGE_CAPACITY` errors when executing queries with `TASK` retry-policy

File: docs/src/main/sphinx/release/release-382.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+* Improve coordinator memory management for task-based fault-tolerant execution.
+  This decreases the chance of queries failing with the
+  `EXCEEDED_TASK_DESCRIPTOR_STORAGE_CAPACITY` error code. ({issue}`12478`)
+
+## BigQuery connector
+
+* Add support for [using cached query results of BigQuery](https://cloud.google.com/bigquery/docs/cached-results).
```
Comment (martint) [inline docs/src/main/sphinx/release/release-382.md:14]: What's the impact? Which kinds of queries would benefit?

File: docs/src/main/sphinx/release/release-382.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+* Support reading wildcard tables. ({issue}`4124`)
+
+## Delta Lake connector
+
+* Support predicate pushdown for `timestamp with time zone` type. ({issue}`12007`)
+* Add support for adding columns with `ALTER TABLE`. ({issue}`12371`)
+
+## Hive connector
+
+* Allow metastore caching without caching partition-level information. This can
```
Comment (martint) [inline docs/src/main/sphinx/release/release-382.md:25]: This is confusing. What does "caching without caching partition-level information" mean?

File: docs/src/main/sphinx/release/release-382.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  be enabled using the `hive.metastore-cache.cache-partitions` catalog
+  configuration property. ({issue}`12343`)
+* Fix queries sometimes failing with `NoSuchElementException: No value present`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-382.md:28]: > Fix potential query failure when metastore caching is enabled.

File: docs/src/main/sphinx/release/release-382.md (hunk 1 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-382.md:30]: If that's the whole list, I would just say:

> Fix query failure when a transactional table contains a column named `operation`, `originalTransaction`, `bucket`, `rowId`, `row`, or `currentTransaction`.

File: docs/src/main/sphinx/release/release-382.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+* Improve `INSERT` performance when data includes `NULL` values. ({issue}`12400`)
+
+## PostgreSQL connector
+
+* Improve `INSERT` performance. ({issue}`12417`)
+
+## Prometheus connector
+
+* Add support for basic authentication. ({issue}`12302`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-382.md:52]: Does this refer to HTTP Basic Authentication? If so, capitalize Basic.

File: docs/src/main/sphinx/release/release-382.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-382.md:5]: PR for that was merged. Thanks

File: docs/src/main/sphinx/release/release-382.md (hunk 7 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-382.md:14]: there is a lot of details to that and they are specific to BigQuery .. the documentation on the Google site goes into that and hence we just link there.

Comment (ebyhr) [reply docs/src/main/sphinx/release/release-382.md:13]: Users can reduce their cost and get result faster. The cache will be reused when the generated SQL matches the past SQL.
